src/transaction.ts

```diff
@@ -183,4 +184,39 @@ export class StacksTransaction extends StacksMessage {
     this.postConditions = LengthPrefixedList.deserialize(bufferReader, PostCondition);
     this.payload = Payload.deserialize(bufferReader);
   }
+
+  /**
+   * Broadcast the signed transaction to a core node
+   *
+   * @param {String} apiUrl - specify the core node URL to broadcast to
+   *
+   * @returns {Promise} that resolves to a response if the operation succeeds
+   */
+  broadcast(apiUrl?: string) {
+    const tx = this.serialize();
+
+    const requestHeaders = {
+      Accept: 'application/json',
+      'Content-Type': 'application/octet-stream',
+    };
+
+    const options = {
+      method: 'POST',
+      headers: requestHeaders,
+      body: tx,
+    };
+
+    const url = apiUrl || `${DEFAULT_CORE_NODE_API_URL}/v2/transactions`;
+
+    return fetchPrivate(url, options).then(response => {
+      if (response.ok) {
+        return response.text();
+        // return response.json();
+      } else if (response.status === 400) {
+        throw new Error('Transaction rejected');
```

Comment:
The error response should be returned to the sender. This would help "debugging" transactions.

Reply:
Transactions can fail in so many different ways -- I think it will be very common for consumers to check the result for a myriad of different rejection cases. Try/catch statements and inspecting the `any` error object will likely be cumbersome and devs often don't perform much error handling like this in general. 

I propose this function return something like:
```ts
type BroadcastResult = 
  | { ok: <txid and whatever else> }
  | { error: { reason: <string enum>, message: string };
```